ktest/testharness/kubeapiserver.go

```diff
@@ -0,0 +1,213 @@
+package testharness
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/kubebuilder-declarative-pattern/ktest/httprecorder"
+	"sigs.k8s.io/yaml"
+)
+
+type TestKubeAPIServer struct {
+	t          *testing.T
+	ctx        context.Context
+	restConfig *rest.Config
+	client     client.Client
+}
+
+func NewTestKubeAPIServer(t *testing.T, ctx context.Context, env *envtest.Environment) *TestKubeAPIServer {
+	s := &TestKubeAPIServer{t: t, ctx: ctx}
+
+	if env != nil {
+		restConfig, err := env.Start()
+		if err != nil {
+			t.Fatalf("failed to start envtest kube-apiserver: %v", err)
+		}
+		s.restConfig = restConfig
+		t.Cleanup(func() {
+			if err := env.Stop(); err != nil {
+				t.Errorf("error stopping envtest: %v", err)
+			}
+		})
+	} else {
+		// Removed for now...
+		t.Fatalf("mockkubeapiserver is not supported with this test harness")
```

Comment:
Are you planning to add this back? If not, I suggest making the parameter non-nil and just getting rid of the branch here.

Reply:
I added it back.  The output is slightly different because of the api discovery v2 features added in k/k, but I figure we can tackle that separately (if we choose to)